### DISCUSSION TITLE
Change water ride lift-hills to move at a constant speed instead of speed being based on boat weight

### DISCUSSION THREAD

**Discussion by DiamondDaveL** [OpenRCT2/OpenRCT2#20180] [Feature requests]
This non-realistic behavior has always been strange to me. Log flume, Splash boats, and Rapids rides have a weird behavior where the "lift" speed is inversely proportional to the number of peeps in a log/boat/raft. 

How should a modification be handled? Is there a mechanical reason to keep the current behavior as default? 

This discussion is similar- https://github.com/OpenRCT2/OpenRCT2/discussions/18294

**Comment by ocalhoun6** [comment]
The rollers on their lift hills don't have positive lockup with the ride vehicles -- they're only propelled upward by friction. So it makes sense to me that full boats (heavy boats) would be slower, since gravity is pulling them backward also, so it's a '2 steps forward 1 step backward' kind of deal trying to get a full boat up a lift.

I *do* wish there were higher speeds available, though. Because currently, it's just *painfully* slow for a full boat to make its way up a large lift.

**Reply by DiamondDaveL** [reply]
With my current knowledge of water ride lift systems, I don't think that behavior is realistic. I imagine the whole conveyor system might slow down if heavier boats are on it, but that's not how it works in the game, each boat moves individually. \
Are the lifts in the game non-connected rollers with individual motors? I am unfamiliar with a boat system that uses that aside from tire drive lifts, which I guess are similar to the game behavior. 

I thought the lifts in-game were modeled after conveyor lifts (just without moving sprites due to the complexity), where they are not individual rollers but a connected belt system. With a connected belt system, all boats would/should move at the same speed. 

Thank you for the discussion.

**Comment by 733737** [comment]
I don't think it is relevant to gameplay to justify being able to increase the lift speed based upon realism. That's not to say we should have 255km/h lifts without cheats but the current vanilla implementation should be replaced with a variable lift speed which defaults to vanilla value similar to the way speedy block brakes were implemented to preserve preexisting designs and prevent breaking td6 compatibility, imo.

**Reply by DiamondDaveL** [reply]
I agree with this reasoning. Maybe that variable lift speed value is a multiplier of the existing calculation, instead of replacing with constant speed, traditional lifts?

**Comment by 733737** [comment]
I thought @karst wanted to replace them with the coaster style lifts entirely, what is the benefit to keeping the current system which leads to boat jams?

**Reply by DiamondDaveL** [reply]
I don't see a benefit to that, but it may be an easier change. That's just speculation, which I shouldn't do. 

I do not see a reason to keep the behavior current vs having a constant lift speed across boats. I was hoping this discussion might reach someone that has an argument for keeping the behavior current/vanilla.

**Comment by cheweytoo** [comment]
> This non-realistic behavior has always been strange to me. Log flume, Splash boats, and Rapids rides have a weird behavior where the "lift" speed is inversely proportional to the number of peeps in a log/boat/raft.

I'm with you here. I'd understand if the speed _of the whole lift hill_ depended on load (which I'm sure happens in real life), but I have yet to encounter an actual water ride with a non-conveyor-belt lift system – which is the only way different per-boat lifting speeds are possible.

Changing the behaviour to modifying the whole lift hill speed would also coincidentally fix https://github.com/OpenRCT2/OpenRCT2/issues/2153 and friends.

**Comment by J-PIE-314** [comment]
PR https://github.com/OpenRCT2/OpenRCT2/pull/20258 is intended to add support for changing water ride lift speeds.
